recipes/isl/all/conanfile.py

```diff
@@ -62,72 +60,68 @@ def _settings_build(self):
-        conf_args = [
-            "--with-int={}".format(self.options.with_int),
-            "--enable-portable-binary",
-            "--enable-shared={}".format(yes_no(self.options.shared)),
-            "--enable-static={}".format(yes_no(not self.options.shared)),
-        ]
+        get(self, **self.conan_data["sources"][self.version],
+                  strip_root=True, destination=self.source_folder)
+
+    def generate(self):
+        tc = AutotoolsToolchain(self)
+        tc.configure_args.append(f'--with-int={self.options.with_int}')
+        tc.configure_args.append("--enable-portable-binary")
+        tc.configure_args.append(f'--enable-shared={"yes" if self.options.shared else "no"}')
+        tc.configure_args.append(f'--enable-static={"yes" if not self.options.shared else "no"}')
         if self.options.with_int == "gmp":
-            conf_args.extend([
-                "--with-gmp=system",
-                "--with-gmp-prefix={}".format(self.deps_cpp_info["gmp"].rootpath.replace("\\", "/")),
-            ])
-        if self.settings.compiler == "Visual Studio":
-            if tools.Version(self.settings.compiler.version) >= 15:
-                self._autotools.flags.append("-Zf")
-            if tools.Version(self.settings.compiler.version) >= 12:
-                self._autotools.flags.append("-FS")
-        self._autotools.configure(args=conf_args, configure_dir=self._source_subfolder)
-        return self._autotools
+            tc.configure_args.append("--with-gmp=system")
+            tc.configure_args.append(f'--with-gmp-prefix={unix_path(self, self.dependencies["gmp"].package_folder)}')
+        if self.settings.compiler == "msvc":
+            if Version(self.settings.compiler.version) >= 15:
+                tc.extra_cflags = ["-Zf"]
+            if Version(self.settings.compiler.version) >= 12:
+                tc.extra_cflags = ["-FS"]
+        env = tc.environment()
+        if self.settings.compiler == "msvc":
+            env.define("AR", f'{unix_path(self, self.dependencies["automake"].ar_lib)} lib')
+            env.define("CC", f'{unix_path(self, self.dependencies["automake"].compile)} cl -nologo {self.settings.compiler.runtime}')
+            env.define("CXX", f'{unix_path(self, self.dependencies["automake"].compile)} cl -nologo {self.settings.compiler.runtime}')
+            env.define("NM", "dumpbin -symbols")
+            env.define("OBJDUMP", ":")
+            env.define("RANLIB", ":")
+            env.define("STRIP", ":")
+        tc.generate(env)
 
     def build(self):
-        with self._build_context():
-            autotools = self._configure_autotools()
-            autotools.make()
+        # Support building with source from Git reop
+        with chdir(self, self.source_folder):
+            command = "./autogen.sh"
+            if os.path.exists(command) and not os.path.exists("configure"):
+                self.run(command)
+        apply_conandata_patches(self)
```

Comment:
```suggestion
```

there is no patch

Reply:
Back ported fix from 0.25 so now apply_conandata_patches is needed.